### ISSUE TITLE
Travis CI

### ISSUE THREAD

**Issue by waynenilsen** [ggpy2/ggpy2#5]
First step in CI setup should be some basic plots should not raise exception with a matrix of python versions and compatible numpy/pandas versions

**Comment by mschmidtesd** [comment]
While putting together a quick and dirty Travis configuration, I took a look at the tests subdirectory.
Oh, what a mess. The existing tests seem to be quite useful for finding further incompatibilities with current versions of the dependencies, but the test code has obviously not been touched in years and is not Python 3 compatible. And the complete test suite is based on visual checks so it's not suited for automated testing.

**Comment by waynenilsen** [comment]
>  the complete test suite is based on visual checks so it's not suited for automated testing.

I don't think that is necessarily true, at least it is running some code and making sure it doesn't raise any exceptions. This is obviously a very low bar for automated testing but maybe somewhere to start.

Failing elements in the matrix are actually great and mean that they can be patched and re-tested.

**Comment by mschmidtesd** [comment]
@waynenilsen you're definitely right, but that was not what I meant. It seems I have compressed the facts too much so that my sentence has lost its original meaning. The tests open a lot of TK windows showing graphs, each of them blocking test execution until the window is manually closed. I don't know how the Travis environment handles this, but it could be a show stopper.

On the other hand I had seen the a part of code observing an environment variable called GGPLOTDEV. Perhaps it could be used to redirect outputs to PNG files and suppress the windows.

**Comment by waynenilsen** [comment]
Ah yes, I probably should have looked at the code, I see what you mean now. Looks like tests folder will need to be reworked significantly.

You may have seen the other repository that @has2k1 has linked in the a different issue [plotnine](https://github.com/has2k1/plotnine). It looks to me like all of this work was essentially done a year ago along with many many other things.

@has2k1 has changed the license on this repo but that my not be an issue as it looks like they are the [primary contributor now](https://github.com/has2k1/plotnine/graphs/contributors). License issues always confuse me and I am certainly not an expert.

**Comment by mschmidtesd** [comment]
Now I'm confused. I had already heard of plotnine, even used it in one of my projects, and I thought, it was a reimplementation or fork of ggpy, but when I look at the github history, it seems to be the other way around. Yhat employees seem to have contributed to plotnine and then in 2016 have started their own project as a rewrite.

**Comment by waynenilsen** [comment]
This is an interesting story.. The [history](https://github.com/has2k1/plotnine/commits/master/LICENSE) of the license file is interesting as well. 

At any rate, I'm not sure there is any reason for this repo to continue to exist as plotnine seems to be superior in essentially every way to the old ggpy code.

**Comment by mschmidtesd** [comment]
Yeah. I thought, keeping ggpy from dying and preventing a monoculture is a good thing, but the more I look into it, the more it seems like riding a dead horse.

**Comment by has2k1** [comment]
@mschmidtesd, [About plotnine](http://plotnine.readthedocs.io/en/stable/about-plotnine.html).

**Comment by mschmidtesd** [comment]
@has2k1 Thanks for the link. Interesting read and the decisions described were definitively right. I come from GNU R and ggplot2, so plotnine is what I#m going with from now on. Just have not been aware of it before.

@waynenilsen  The bar for improvements in plotnine is a bit higher, but I think we should rather contribute to it than resurrecting the dead horse named ggpy.
